### PR TITLE
Write Foundry Guide and update Solidity Style Guide

### PR DESCRIPTION
**Pull request by RensR**

### CODE REVIEW

File: contracts/FOUNDRY_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+We will assume the profile is called `newproject`.
+
+The foundry profile should look similar to this.
+
+```toml
+[profile.newproject]
+solc_version = '0.8.24'
+src = 'src/v0.8/newproject'
+test = 'src/v0.8/newproject/test'
+optimizer_runs = 1_000_000
```
Comment (matYang) [inline contracts/FOUNDRY_GUIDE.md:23]: should this match `OPTIMIZE_RUNS=26000` in native compilation script?

File: contracts/STYLE_GUIDE.md [hunk 2]
```diff
@@ -6,11 +6,19 @@ Rules are all enforced through CI, this can be through Solhint rules or other to
 
 ## Background
 
-Our starting point is the [official Solidity Style Guide](https://docs.soliditylang.org/en/v0.8.21/style-guide.html) and [ConsenSys's Secure Development practices](https://consensys.github.io/smart-contract-best-practices/), but we deviate in some ways. We lean heavily on [Prettier](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/.prettierrc) for formatting, and if you have to set up a new Solidity project we recommend starting with [our prettier config](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/.prettierrc). We are trying to automate as much of this style guide with Solhint as possible.
+Our starting point is the [official Solidity Style Guide](https://docs.soliditylang.org/en/v0.8.21/style-guide.html) and
+[ConsenSys's Secure Development practices](https://consensys.github.io/smart-contract-best-practices/), 
+but we deviate in some ways.
+We lean heavily on [Prettier](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/.prettierrc) for formatting.
```
Comment (matYang) [inline contracts/STYLE_GUIDE.md]: Leaning heavily on Prettier is no longer the case right

**Comment by matYang** [comment]
nice foundry starter guide =)

File: contracts/FOUNDRY_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+with fewer merge conflicts.
+
+Examples of correct test names for a function `getFee` are:
+
+```
+test_getFee - the base success case
+test_getFee_MultipleFeeTokens - another success case with a specific scenario
+test_getFee_RevertWhen_CursedByRMN - getFee reverts when it's cursed by the RMN. The error name should be used as condition when there is a single tests that checks for it
+testFuzz_getFee_OnlyFeeTokens - a fuzz test that asserts that only fee tokens are used
+testFork_getFee_UniswapV3MainnetFee - a fork test tha tuses Uniswap V3 on mainnet to get the fee
```
Comment (jhweintraub) [inline contracts/FOUNDRY_GUIDE.md]: typo

**Review by matYang — APPROVED** [review]

File: contracts/FOUNDRY_GUIDE.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+
+### Test naming
+
+Tests are named according to the following format:
+
+```
+test_FunctionName_Description for standard tests.
+test_FunctionName_RevertWhen_Condition for tests expecting a revert.
+testFuzz_FunctionName_Description for fuzz tests.
+testFork_FunctionName_Description for tests that fork from a network.
```
Comment (RayXpub) [inline contracts/FOUNDRY_GUIDE.md:183]: nit: shouldn't be `test_functionName_Description`?

**Review by RayXpub — APPROVED** [review]

**Pull request merged**